5-design-patterns/homework/src/Shipment.ts

```diff
@@ -0,0 +1,24 @@
+import { ShipmentParams} from "./interfaces";
+import { Shipper } from "./Shipper";
+
+export abstract class Shipment {
+    protected shipmentParams: ShipmentParams;
+    private shipmentId: number;
+    protected shipper: Shipper;
+    private shipmentIdMock = 0;
+
+    constructor(shipmentParams: ShipmentParams, shipper: Shipper, shipmentId?: number) {
```

Comment:
I'm not sure about injecting shipper and shipment id. Should client of class that extends this one create a right shipper instance for own - it's a question.